src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ByValueContentsMarshalKindValidator.cs

```diff
@@ -29,58 +27,28 @@ public ResolvedGenerator Create(TypePositionInfo info, StubCodeContext context)
 
         private static ResolvedGenerator ValidateByValueMarshalKind(TypePositionInfo info, StubCodeContext context, ResolvedGenerator generator)
         {
-            if (generator.Generator is Forwarder)
+            if (generator.Generator is Forwarder || info.ByValueContentsMarshalKind == ByValueContentsMarshalKind.Default)
             {
                 // Forwarder allows everything since it just forwards to a P/Invoke.
```

Comment:
```suggestion
                // Forwarder allows everything since it just forwards to a P/Invoke.
                // The Default marshal kind is always valid.
```